go/apps/demo/demo_client/demo_client.go

```diff
@@ -142,28 +135,64 @@ func main() {
 	} else {
 		serverAddr = strings.TrimPrefix(serverEnvVar, "tcp://")
 		if serverAddr == serverEnvVar {
-			glog.Fatalf("client: invalid SERVER_PORT environment variable value '%s'\n", serverEnvVar)
+			usage("client: invalid SERVER_PORT environment variable value '%s'\n", serverEnvVar)
 		}
 	}
 
 	switch *demoAuth {
 	case "tcp", "tls", "tao":
 	default:
-		glog.Fatalf("unrecognized authentication mode: %s\n", *demoAuth)
+		usage("unrecognized authentication mode: %s\n", *demoAuth)
 	}
 
-	glog.Info("Go Tao Demo")
+	fmt.Println("Go Tao Demo Client")
 
 	if tao.Parent() == nil {
-		glog.Fatal("can't continue: No host Tao available")
+		fail(nil, "can't continue: No host Tao available")
+	}
+
+	domain, err := tao.LoadDomain(configPath(), nil)
+	failIf(err, "error: couldn't load the tao domain from %s\n", configPath())
+
+	doClient(domain)
+	fmt.Println("Client Done")
+}
+
+func domainPath() string {
+	if *domainPathFlag != "" {
+		return *domainPathFlag
+	}
+	if path := os.Getenv("TAO_DOMAIN"); path != "" {
+		return path
+	}
+	usage("Must supply -tao_domain or set $TAO_DOMAIN")
+	return ""
+}
+
+func configPath() string {
+	return path.Join(domainPath(), "tao.config")
+}
+
+func failIf(err error, msg string, args ...interface{}) {
+	if err != nil {
+		fail(err, msg, args...)
 	}
+}
 
-	domain, err := tao.LoadDomain(*configPath, nil)
+func fail(err error, msg string, args ...interface{}) {
+	s := fmt.Sprintf(msg, args...)
 	if err != nil {
-		glog.Fatalf("error: couldn't load the tao domain from %s\n", *configPath)
+		fmt.Fprintf(os.Stderr, "%v: %s\n", err, s)
+	} else {
+		fmt.Fprintf(os.Stderr, "error: %s\n", s)
 	}
+	os.Exit(2)
+}
 
-	doClient(domain)
-	glog.Info("Client Done")
-	glog.Flush()
+func usage(msg string, args ...interface{}) {
+	s := fmt.Sprintf(msg, args...)
+	fmt.Fprintf(os.Stderr, "%s\n", s)
```

Comment:
I must be missing something here: why isn't this combined with the line above? Fprintf does the same string/vargs formatting as Sprintf.